src/main/java/org/folio/service/inventory/InventoryHoldingManager.java

```diff
@@ -317,4 +320,43 @@ public Future<String> handleHoldingsRecord(CompositePoLine compPOL, Location loc
     }
   }
 
+  public Future<List<JsonObject>> getHoldingsForAllTenants(CompositePoLine poLine, RequestContext requestContext) {
+    var futures = getHoldingsFutures(poLine, requestContext);
+    return GenericCompositeFuture.all(new ArrayList<>(futures.values()))
+      .map(ar -> futures.values().stream()
+        .flatMap(future -> future.result().stream())
+        .toList()
+      );
+  }
+
+  public Map<String, Future<List<JsonObject>>> getHoldingsFutures(CompositePoLine poLine, RequestContext requestContext) {
+    String currentTenantId = TenantTool.tenantId(requestContext.getHeaders());
+    Map<String, List<String>> holdingsByTenant = poLine.getLocations()
+      .stream()
+      .filter(location -> location.getHoldingId() != null)
+      .collect(Collectors.groupingBy(
+        location -> ObjectUtils.defaultIfNull(location.getTenantId(), currentTenantId),
+        Collectors.mapping(Location::getHoldingId, Collectors.toList())
+      ));
+
+    return holdingsByTenant.entrySet()
+      .stream()
+      .collect(Collectors.toMap(Map.Entry::getKey, entry -> getHoldingsFuture(entry.getKey(), entry.getValue(), requestContext)));
+  }
+
+  private Future<List<JsonObject>> getHoldingsFuture(String tenantId, List<String> holdingIds, RequestContext requestContext) {
```

Comment:
Suggestion: These method names should be improved. maybe it can be same name with different attributes as overload methos. 
tenantHoldings.getKey, tenantHoldings.getValue() is more clear we are passing tenant and holsing

Reply:
Renamed suggested methods and variables